expression/builtin_time.go

```diff
@@ -4856,7 +4856,7 @@ func (b *builtinUnixTimestampIntSig) evalIntWithCtx(ctx sessionctx.Context, row
 	}
 
 	tz := ctx.GetSessionVars().Location()
-	t, err := val.GoTime(tz)
+	t, err := val.AdjustedGoTime(tz)
```

Comment:
I think expression/builtin_time_vec.go will need modifying too?

Reply:
Good catch! I added the fix there too.